include/wil/cppwinrt_helpers.h

```diff
@@ -212,6 +212,48 @@ namespace wil::details
 #endif // __WIL_CPPWINRT_MICROSOFT_UI_DISPATCHING_HELPERS
 /// @endcond
 
+#if defined(WINRT_Windows_Foundation_H) && !defined(__WIL_CPPWINRT_WINDOWS_FOUNDATION_HELPERS)
+#define __WIL_CPPWINRT_WINDOWS_FOUNDATION_HELPERS
+namespace Windows::Foundation
+{
+    /// @cond
+    struct IMemoryBufferByteAccess;
+    /// @endcond
+}
+
+namespace wil
+{
+    //! Returns a view into the underlying bytes of a memory buffer
+    //! provided in the form of an IMemoryBufferReference.
+    //! The caller is responsible for ensuring that the memory buffer's
+    //! lifetime encompasses the lifetime of the returned view.
+    //! By default, returns an array_view<uint8_t>, but you can provide an alternate
+    //! type such as to_array_view<double>.
+    //! You must include memorybuffer.h in order to use this overload of to_array_view.
+    template<typename T = uint8_t, int V = 0>

```

Comment:
What's the `V` here for? Just to prevent compilation errors when `IMemoryBufferByteAccess` is not defined?

Reply:
Yes, the V is to prevent the template from being instantiated before IMemoryBufferByteAccess is defined.

Reply:
Yeah, was about to ask if the `is_same_v<T, T>` trick didn't delay evaluation for some reason